api/utils/marshamallow_schemas.py

```diff
@@ -0,0 +1,103 @@
+"""Validation schemas."""
+from marshmallow import (Schema, fields, post_load, validate,
+                         ValidationError)
+from api.models import User, Activity, ActivityType
+
+
+class BaseSchema(Schema):
+    """Creates a base validation schema."""
+
+    uuid = fields.String(dump_only=True, validate=[
+        validate.Length(max=36)])
+    name = fields.String(
+        required=True,
+        error_messages={
+            'required': {'message': 'A name is required.',
+                         'status_code': 400}
+        })
+    photo = fields.String()
+    created_at = fields.DateTime(dump_only=True)
+    modified_at = fields.DateTime(dump_only=True)
+    description = fields.String()
+
+    @post_load
+    def verify_activity_type(self, data):
+        """Verify if activity type exists."""
+        activity_type = ActivityType.query.filter(
+            ActivityType.uuid.ilike(data['activity_type_id'])).first()
+        existing_activity = Activity.query.filter(
+            Activity.name.ilike(data['name'])).first()
```

Comment:
This might not be the best way of checking duplicate activities. Activities might have similar text but be different activities altogether. Maybe ask the user to confirm the addition since a similar activity already exists.

Reply:
This we looked into so it should be ok.